src/symbolic/symbolic.jl

```diff
@@ -228,6 +228,33 @@ isinvolution(::AbstractAlgebra, ::typeof(-)) = true
 isinvolution(::AbstractAlgebra, ::typeof(inv)) = true
 
 
+
+iscollapsable(alg) = (f) -> iscollapsable(alg, f)
+iscollapsable(alg, f::FinchNode) = f.kind === literal && iscollapsable(alg, f.val)
+"""
+    iscollapsable(algebra, f)
+
+Return true when `f` is collapsable in loop reduction.
+"""
+iscollapsable(::Any, f) = false
+iscollapsable(::AbstractAlgebra, ::typeof(or)) = true
+iscollapsable(::AbstractAlgebra, ::typeof(and)) = true
+iscollapsable(::AbstractAlgebra, ::typeof(|)) = true
+iscollapsable(::AbstractAlgebra, ::typeof(&)) = true
+iscollapsable(::AbstractAlgebra, ::typeof(+)) = true
+iscollapsable(::AbstractAlgebra, ::typeof(*)) = true
+iscollapsable(::AbstractAlgebra, ::typeof(min)) = true
+iscollapsable(::AbstractAlgebra, ::typeof(max)) = true
```

Comment:
I think `iscollapsable` and `collapsed` should be the same function. `collapsed` should have the signature `collapsed(algebra, ::f, idx, ext, node)` and return `nothing` if not collapsible, or the collapsed form otherwise. Giving `collapsed` these arguments would allow you to write

```
collapsed(algebra, ::typeof(+), idx, ext::ContinuousExtent, node) = 
   if @capture node :call(*, ~a1..., :call(d, idx), ~a2...)
       :call(*, a1..., a2..., measure(ext))
   end
   ```

Reply:
At this point, I think if you remove iscollapsible the code will behave the same.